### PR TITLE
Issue 254 clean up spme outputs

### LINKED ISSUE #254: Clean up SPMe outputs
**Summary**
Clean up SPMe outputs so that large expressions are abstracted into submodels

 **Motivation**
Simplify model files and also allows for a more maintainable code if the large expressions are stored within a submodel instead of being in every model.

### PR DESCRIPTION
**Pull request by Scottmar93**
# Description
Cleaned up SPMe output by implementing appropriate equations with submodels. Also written a standard set of output variables which we can ensure is consistent across all model for a specific chemistry.

Fixes #254

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by martinjrobins** [comment]
yea, I think users are all going to be interested to calculate and plot different things. You could have a small number of standard output variables. in the `pybamm.LithiumIonBaseModel` or `pybamm.BaseModel` class. Just put the name in the dict equated to `None`, then have the derived classes replace that with the actual expression.

**Comment by tinosulzer** [comment]
Refactor using {"variable": None} dictionary in the base model

**Comment by Scottmar93** [comment]
Thanks for the feedback on this I think it is much better now. I have added standard variables to base_model as suggested. I have also put a check in check_model_wellposedness to checks if these variables have been added. I also went ahead and did the DFN outputs aswell which was pretty simple to do.

File: pybamm/models/base_models.py [hunk 1]
```diff
@@ -365,6 +365,16 @@ def check_well_posedness(self, post_discretisation=False):
...
 
+        # Standard Output Variables
+        for output, expression in self._variables.items():
+            if expression is None:
+                raise pybamm.ModelError(
+                    """The standard output variable '{}' which is
+                    required for testing has not been supplied.""".format(
+                        output
+                    )
+                )
```
Comment (tinosulzer) [inline pybamm/models/base_models.py]: can you add a test for this (to check it fails as expected?)

**Pull request merged**